[SYCL] Add SYCL_HOST_UNIFIED_MEMORY environment variable (#3294)
intel/llvm: sycl/doc/EnvironmentVariables.md

| SYCL_PARALLEL_FOR_RANGE_ROUNDING_TRACE | Any(\*) | Enables tracing of parallel_for invocations with rounded-up ranges. |
| SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING | Any(\*) | Disables automatic rounding-up of parallel_for invocation ranges. |
| SYCL_ENABLE_PCI | Integer | When set to 1, enables obtaining the GPU PCI address when using the Level Zero backend. The default is 0. |

> | SYCL_HOST_UNIFIED_MEMORY | Integer | Force the execution graph builder to behave as if no non-host devices support host unified memory when set to 0, or as if all of them do when set to 1. |

Reviewer: ```suggestion
| SYCL_HOST_UNIFIED_MEMORY | Integer | Enforce host unified memory support/unsupport for the execution graph builder. If set to 0 it is enforced as not supported by all devices. If set to 1 it is enforced as supported by all devices. |
```

Reviewer 2: As a variant:
`Control host unified memory support for the execution graph builder. Setting to 0/1 forces the graph builder to unsupport/support the host USM on all devices.`
